### PR TITLE
[ZEPPELIN-5185]. HiveMonitor thread is never finished sometimes

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?

Sometimes, hive monitor thread will never exit when getting the following exception (see screenshot) 
image

This PR would exit the monitor thread when any exception happens and also introduce property `zeppelin.jdbc.hive.monitor.query_interval` to control the query interval. 

### What type of PR is it?
[Bug Fix]

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-5185

### How should this be tested?
* CI pass

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: jdbc/src/main/java/org/apache/zeppelin/jdbc/hive/HiveUtils.java [hunk 1]
```diff
@@ -71,11 +71,14 @@ public static void startHiveMonitorThread(Statement stmt,
...
     final long timeoutThreshold = Long.parseLong(
             jdbcInterpreter.getProperty("zeppelin.jdbc.hive.timeout.threshold", "" + 60 * 1000));
+    final long queryInterval = Long.parseLong(jdbcInterpreter.getProperty("zeppelin.jdbc.hive.monitor.query_interval",
+            DEFAULT_QUERY_PROGRESS_INTERVAL + ""));
     Thread thread = new Thread(() -> {
       boolean jobLaunched = false;
       long jobLastActiveTime = System.currentTimeMillis();
       while (hiveStmt.hasMoreLogs() && !Thread.interrupted()) {
         try {
+          Thread.sleep(queryInterval);
```
Comment (Reamer) [inline jdbc/src/main/java/org/apache/zeppelin/jdbc/hive/HiveUtils.java:81]: Maybe we should catch the Interruptexception in a separate 'catch' to interrupt the current thread and to log a better message.

Comment (zjffdu) [reply jdbc/src/main/java/org/apache/zeppelin/jdbc/hive/HiveUtils.java:81]: Fixed

Comment (Reamer) [reply jdbc/src/main/java/org/apache/zeppelin/jdbc/hive/HiveUtils.java:81]: For good practice we should restore the interrupted state.
Take a look at this rule:
https://rules.sonarsource.com/java/tag/multi-threading/RSPEC-2142

Maybe it makes sense to swap the ’while’ and ’try’ to reduce the amount of ’break’.

Comment (zjffdu) [reply jdbc/src/main/java/org/apache/zeppelin/jdbc/hive/HiveUtils.java:81]: Good idea, fixed

Comment (Reamer) [reply jdbc/src/main/java/org/apache/zeppelin/jdbc/hive/HiveUtils.java:81]: `hiveStmt.isClosed())` should be part of the while condition.

**Comment by Reamer** [comment]
LGTM

**Pull request closed without merging**